app/components/profile.jsx

```diff
@@ -19,73 +23,60 @@ export class Profile extends Component {
     this.reset();
   }
 
-  componentWillUpdate (props) {
-    this.reset(props);
+  componentWillUpdate ({ params: { username } }) {
+    if (username !== this.props.params.username) {
```

Comment:
since we havent destructured `nextProps` into `{ params: { username } }`, we no longer have access to `username` directly anymore (your linter will complain about it). so we just need to access it from `nextProps`.

final result:

```js
if (nextProps.params.username !== this.props.params.username) {
```

you should also do this in `tracker.jsx` (line 29)